bpf-map-pressure-exporter/src/fetcher.go

```diff
@@ -0,0 +1,181 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/cilium/ebpf"
+	"github.com/cybozu-go/log"
+)
+
+type IBpfMapPressureFetcher interface {
+	GetMetrics() []bpfMapPressure
+	Start(context.Context)
+}
+
+type bpfMapPressureFetcher struct {
+	mapNameStrings []string
+	fetchInterval  time.Duration
+	mutex          sync.RWMutex
+	metrics        []bpfMapPressure
+}
+
+type bpfMapPressure struct {
+	mapId       uint32
+	mapName     string
+	mapPressure float64
+}
+
+func newFetcher(mapNameStrings []string, interval time.Duration) *bpfMapPressureFetcher {
+	return &bpfMapPressureFetcher{
+		mapNameStrings: mapNameStrings,
+		fetchInterval:  interval,
+	}
+}
+
+func (f *bpfMapPressureFetcher) Start(ctx context.Context) {
+	f.update()
+
+	ticker := time.NewTicker(f.fetchInterval)
+	for {
+		select {
+		case <-ctx.Done():
+			ticker.Stop()
+			return
+		case <-ticker.C:
+			f.update()
+		}
+	}
+}
+
+func (f *bpfMapPressureFetcher) update() {
+	metrics := []bpfMapPressure{}
+
+	defer func() {
+		f.mutex.Lock()
+		defer f.mutex.Unlock()
+		f.metrics = metrics
+	}()
+
+	metrics = f.fetch()
```

Comment:
```suggestion
	metrics = f.fetch()
	f.mutex.Lock()
	defer f.mutex.Unlock()
	f.metrics = metrics
```

nit: I'd write like these :)

Reply:
https://github.com/cybozu/neco-containers/pull/1155/commits/13b4764764a5c0476d085cdb92880a1432d7d924